boards/Makefile.common

```diff
@@ -5,16 +5,17 @@ MAKEFLAGS += -R
 
 MAKEFILE_COMMON_PATH := $(dir $(abspath $(lastword $(MAKEFILE_LIST))))
 
-TOOLCHAIN ?= arm-none-eabi
+TOOLCHAIN ?= llvm
 
 CARGO ?= cargo
-# This will hopefully move into Cargo.toml (or Cargo.toml.local) eventually
-RUSTFLAGS_FOR_CARGO_LINKING := "-C link-arg=-Tlayout.ld -C linker=lld -Z linker-flavor=ld.lld -C relocation-model=dynamic-no-pic"
 
-SIZE      ?= $(TOOLCHAIN)-size
-OBJCOPY   ?= $(TOOLCHAIN)-objcopy
-OBJDUMP   ?= $(TOOLCHAIN)-objdump
-OBJDUMP_FLAGS += --disassemble-all --source --disassembler-options=force-thumb -C --section-headers
+# This will hopefully move into Cargo.toml (or Cargo.toml.local) eventually.
+# lld uses the page size to align program sections. It defaults to 4096 and this
+# puts a gap between before the .relocate section. `zmax-page-size=512` tells
+# lld the actual page size so it doesn't have to be conservative.
+RUSTFLAGS_FOR_CARGO_LINKING := "-C link-arg=-Tlayout.ld -C linker=lld \
+-Z linker-flavor=ld.lld -C relocation-model=dynamic-no-pic \
+-C link-arg=-zmax-page-size=512"
```

Comment:
Is this page size the flash page size? Isn't this chip-specific? (Do all our chips happen to have 512 byte pages right now?)

Reply:
I think so, but exists so that something can configure particular settings on particular flash pages, which we don't do. We need it set to 512 (or smaller) to match this: https://github.com/tock/tock/pull/1031/files#diff-d5b6e65f5bfb42e94ee9772ef4ac3b07L143 so that the relocate section is placed after text without a gap.

Reply:
Ah, makes sense. I think we'll probably have to come up with something for configurable page sizes some day, but that doesn't feel like a today problem.

I wish the github UI had some way of marking a comment as "resolved" so it'd collapse automatically.